File: bindings/python/src/kortecx/secrets.py
"""Operator secret-store admin (D170 / MM-3) — the named secret VALUES the runtime
holds for connector credentials + trigger auth, surfaced by ``PutSecret`` /
``ListSecretNames`` / ``DeleteSecret``.

Kept in its own module (the feedback.py / module-per-concern precedent, GR3). The
secret VALUE never crosses back over the wire (D81): ``ListSecretNames`` returns
only NAMES + audit timestamps. A connector ``credential_ref`` / a trigger
``auth_secret_ref`` NAMES one of these rows; the value resolves server-side at
dial / fire time, never in the client.
"""

from __future__ import annotations

from dataclasses import dataclass
from typing import List

from .v1 import gateway_pb2 as _g


@dataclass(frozen=True)
class SecretName:
    """One stored secret's metadata (``ListSecretNames``) — NAME + audit
    timestamps only. The secret VALUE is never on this wire (D81); the timestamps
    are display/audit wall clock (never identity, never a hash input)."""

    name: str
    created_unix_ms: int
    updated_unix_ms: int

    @classmethod
    def from_proto(cls, s: "_g.SecretName") -> "SecretName":
        return cls(
            name=s.name,
            created_unix_ms=s.created_unix_ms,
            updated_unix_ms=s.updated_unix_ms,
        )


@dataclass(frozen=True)
class SecretNamesPage:
    """One ``ListSecretNames`` page (deterministic ``(name)`` order)."""

    names: List[SecretName]
    has_more: bool

File: bindings/typescript/src/secrets.ts
/**
 * The local OS-keychain secret store views (MM-3 / D110). A `SecretRef` NAME is
 * what a connection's / trigger's `credential_ref` points at; the secret VALUE is
 * write-only — it appears ONLY as a `PutSecret` argument and is never returned on
 * any read. `ListSecretNames` surfaces NAMES + audit timestamps only. Kept in its
 * own module (the `alerts.ts`/`toolscout.ts` module-per-concern precedent).
 *
 * SN-8: `PutSecret`/`DeleteSecret` write host credential material and are gated
 * loopback-only + an authenticated party server-side; the SDK only *carries* the
 * value to the handler and *encodes* nothing sensitive on a read.
 */

import type { SecretName as PbSecretName } from "./gen/kortecx/v1/gateway_pb.js";

/** One stored secret's NAME + audit timestamps (`ListSecretNames`). The VALUE is
 *  never on this wire (write-only). `createdUnixMs`/`updatedUnixMs` are audit-only
 *  wall-clocks (off every hash). */
export class SecretNameRow {
  constructor(
    readonly name: string,
    readonly createdUnixMs: number,
    readonly updatedUnixMs: number,
  ) {}

  static fromProto(s: PbSecretName): SecretNameRow {
    return new SecretNameRow(s.name, Number(s.createdUnixMs), Number(s.updatedUnixMs));
  }

  /** A plain snake_case object (stable wire-shaped serialization for UIs/logs). */
  toJSON() {
    return {
      name: this.name,
      created_unix_ms: this.createdUnixMs,
      updated_unix_ms: this.updatedUnixMs,
    };
  }
}

/** One `ListSecretNames` page (deterministic `(name)` order). */
export interface SecretNamesPage {
  readonly names: readonly SecretNameRow[];
  readonly hasMore: boolean;
}
